core/src/test/java/org/elasticsearch/gateway/PrimaryShardAllocatorTests.java

```diff
@@ -175,6 +176,33 @@ public void testStoreException() {
     }
 
     /**
+     * Tests that when the node returns a ShardLockObtainFailedException, it will be considered as a valid shard copy
+     */
+    public void testShardLockObtainFailedException() {
```

Comment:
can we add tests for cases with more shards? for example the case where we "prefer" other shard copies with this exception?